docs/csharp/codedoc.md

```diff
@@ -2,26 +2,1280 @@
+
+```csharp
+/*
+    The main Math class
+    Contains all methods for performing basic math functions
+*/
+/// <summary>
+/// The main Math class.
+/// Contains all methods for performing basic math functions.
+/// </summary>
+/// <remarks>
+/// This class can add, subtract, multiply and divide.
+/// </remarks>
+public class Math
+{
+
+}
+```
+
+### &lt;returns&gt;
+
+As the name suggests you use the `<returns>` tag in the comment for a method declaration to describe its return value.
+Like before this will be illustrated on the first `Add` method go ahead an implement it on other methods.
+
+```csharp
+// Adds two integers and returns the result
+    /// <summary>
+    /// Adds two integers and returns the result.
+    /// </summary>
+    /// <returns>
+    /// The sum of two integers.
+    /// </returns>
+    public static int Add(int a, int b)
+    {
+        // If any parameter is equal to the max value of an integer
+        // and the other is greater than zero
+        if ((a == int.MaxValue && b > 0) || (b == int.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+```
+
+### &lt;value&gt;
+
+The `<value>` works similarly to the `<returns>` tag except that you use it for properties.
+Assuming your `Math` library had a static property called `PI` here's how you'll use this tag:
+
+```csharp
+/*
+    The main Math class
+    Contains all methods for performing basic math functions
+*/
+/// <summary>
+/// The main Math class.
+/// Contains all methods for performing basic math functions.
+/// </summary>
+/// <remarks>
+/// This class can add, subtract, multiply and divide.
+/// </remarks>
+public class Math
+{
+    /// <value>Gets the value of PI.</value>
+    public static double PI { get; }
+}
+```
+
+### &lt;example&gt;
+
+You use the `<example>` tag to include an example in your XML documentation.
+This involves using the child `<code>` tag.
+
+```csharp
+// Adds two integers and returns the result
+    /// <summary>
+    /// Adds two integers and returns the result.
+    /// </summary>
+    /// <returns>
+    /// The sum of two integers.
+    /// </returns>
+    /// <example>
+    /// <code>
+    /// int c = Math.Add(4, 5);
+    /// if (c > 10)
+    /// {
+    ///     Console.WriteLine(c);
+    /// }
+    /// </code>
+    /// </example>
+    public static int Add(int a, int b)
+    {
+        // If any parameter is equal to the max value of an integer
+        // and the other is greater than zero
+        if ((a == int.MaxValue && b > 0) || (b == int.MaxValue && a > 0))
+            throw new System.OverflowException();
+
+        return a + b;
+    }
+```
+
+The `code` tag preserves line breaks and indentation for longer examples.
+
+### &lt;para&gt;
+
+You may find you need to format the content of certain tags and that's where the `<para>` tag comes in.
+You usually use it inside a tag, such as `<summary>`, `<remarks>`, or `<returns>`, and lets you divide text into paragraphs.
```

Comment:
I probably wouldn't recommend multiple paragraphs in the summary tag, because of how that looks in intellisense.